referenceqvm/qam.py

```diff
@@ -91,8 +92,7 @@ def load_program(self, pyquil_program):
 
         # NOTE: all_inst is set by the subclass
         if invalid is True and self.all_inst is False:
-            raise TypeError("In QVM_Unitary, only Gates and DefGates are "
-                            "supported")
+            raise TypeError("Some gates used are not allowed in this QAM")
```

Comment:
will defgates let the user sneak through invalid programs?

Reply:
yes, but that was always a risk of defgates that was weighed in the original quil spec.  If users want to use defgate they should know that it takes them outside of the QAM model.  Remember QAM requires gate specification in the definition.  If the user wants to add something to this list then we are trusting them to know what they're doing.